serving/src/test/java/feast/serving/service/CachedSpecStorageTest.java

```diff
@@ -0,0 +1,115 @@
+package feast.serving.service;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.testing.FakeTicker;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import feast.specs.EntitySpecProto.EntitySpec;
+import feast.specs.FeatureSpecProto.FeatureSpec;
+import feast.specs.StorageSpecProto.StorageSpec;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import org.junit.Before;
+import org.junit.Test;
+
+public class CachedSpecStorageTest {
+
+  private FakeTicker fakeTicker;
+  private CoreService coreService;
+  private CachedSpecStorage cachedSpecStorage;
+
+  @Before
+  public void setUp() throws Exception {
+    fakeTicker = new FakeTicker();
+    coreService = mock(CoreService.class);
+    ListeningExecutorService executorService = MoreExecutors.listeningDecorator(Executors.newSingleThreadExecutor());
+    cachedSpecStorage = new CachedSpecStorage(coreService, executorService,
+        Duration.ofSeconds(5), fakeTicker);
+  }
+
+  @Test
+  public void shouldNotBeNull() {
+    assertNotNull(cachedSpecStorage);
+  }
+
+  @Test
+  public void testPopulateCache() {
```

Comment:
can you add a test for what happens when you exceed the max size of cache in populating. So we're explicit about the intended behaviour.